pkg/manager/manager.go

```diff
@@ -82,34 +83,31 @@ func (k *KubeMacPoolManager) Run(rangeStart, rangeEnd net.HardwareAddr) error {
 		return fmt.Errorf("unable to create a kubernetes client error %v", err)
 	}
 
-	for k.continueToRunManager {
-		log.Info("waiting for manager to become leader")
-		err = k.waitToStartLeading()
-		if err != nil {
-			log.Error(err, "failed to wait for leader election")
-			continue
-		}
-
-		healthProbeHost, ok := os.LookupEnv("HEALTH_PROBE_HOST")
-		if !ok {
-			log.Error(err, "Failed to load HEALTH_PROBE_HOST from environment variable")
-			os.Exit(1)
-		}
+	healthProbeHost, ok := os.LookupEnv("HEALTH_PROBE_HOST")
+	if !ok {
+		log.Error(err, "Failed to load HEALTH_PROBE_HOST from environment variable")
+		os.Exit(1)
+	}
 
-		healthProbePort, ok := os.LookupEnv("HEALTH_PROBE_PORT")
-		if !ok {
-			log.Error(err, "Failed to load HEALTH_PROBE_PORT from environment variable")
-			os.Exit(1)
-		}
+	healthProbePort, ok := os.LookupEnv("HEALTH_PROBE_PORT")
+	if !ok {
+		log.Error(err, "Failed to load HEALTH_PROBE_PORT from environment variable")
+		os.Exit(1)
+	}
 
+	for k.continueToRunManager {
```

Comment:
why do we need `for`?

Reply:
notice [waitForSignal go routine](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/7bffdd3ca7959767a3e10a81d6bf272d98a8fca4/pkg/manager/manager.go#L177). it uses channels to signal stuff like "kubevir tInstalled".

Following the kubevirt example, Should kubevirt not be present during the start of kmp, then all is paused.
but when kubevirt is installed then kubevirtInstalledChannel willbe closed, which will pop up a signal in `waitForSignal` which will cause us to return to the start of the loop and rerun the kmp without a total pod restart..